### PR TITLE
Zypper Package Timeouts support ( #2943 )

### PR DESCRIPTION
**Pull request by gsaslis**
Adding support for timeouts to zypper package provider (issue #2943 )

### CODE REVIEW

**Comment by danielsdeleo** [comment]
Can you add tests for the ZypperPackage resource and fix up the test failures travis reported?

File: lib/chef/resource/zypper_package.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+require 'chef/resource/package'
+require 'chef/provider/package/zypper'
+
+class Chef
+  class Resource
+    class ZypperPackage < Chef::Resource::Package
+
+      provides :zypper_package
+      provides :package, os: "linux", platform_family: [ "suse" ]
```
Comment (btm) [inline lib/chef/resource/zypper_package.rb]: according to lib/chef/provider/provider_mapping.rb, zypper is the default package provider for both "opensuse" and "suse".

**Comment by btm** [comment]
The zypper provider doesn't use allow_downgrade, arch, or flush_cache. It looks like you copied the yum_package resource, but you need to remove the attributes that aren't used by this provider.

As we add provides statements I think it makes sense to remove the corresponding lines from lib/chef/provider/platform_mapping.rb since that is deprecated, like what was done in https://github.com/chef/chef/pull/1462.

See https://www.chef.io/blog/2015/02/10/chef-12-provider-resolver/ for more information on the provider mapping changes.

**Comment by gsaslis** [comment]
wow, thanks for your prompt feedback guys... Yeah, it's my first contribution (attempt ;) ) to chef, so I don't really know all the internals that well... will take a look and revert on the points brought up - thanks again!

**Comment by danielsdeleo** [comment]
@gsaslis Cool, reach out if you need any help.

File: lib/chef/resource/zypper_package.rb (hunk 1 above)
Comment (gsaslis) [inline lib/chef/resource/zypper_package.rb]: I've always been using platform_family 'suse', based on this: https://docs.chef.io/dsl_recipe.html#platform-family

**Comment by gsaslis** [comment]
ok, so i managed to completely break my previous branch during rebase... 
I've now recreated it, but it still won't let me reopen this PR...  : ( 
Sorry guys...

**Comment by btm** [comment]
I think if you push your branch to github under the same name as the old one, this can be reopened. The branch has to exist first.

```gsaslis deleted the gsaslis:zypper_timeouts branch 4 hours ago```

**Comment by gsaslis** [comment]
yeah, that's what i'd tried to do... it does exist... https://github.com/gsaslis/chef/tree/zypper_timeouts

**Comment by gsaslis** [comment]
any ideas?

**Pull request closed without merging**